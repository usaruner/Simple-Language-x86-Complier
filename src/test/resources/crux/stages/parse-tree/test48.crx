// non-recursive fib

func main() : void {
  var i:int;
  let i = 0;
  var a:int;
  var b:int;
  let a = 0;
  let b = 1;
  while (i < 10) {
    ::printInt(b);
    ::println();
    var c:int;
    let c = a + b;
    let a = b;
    let b = c;
    let i = i + 1;
  }
}
