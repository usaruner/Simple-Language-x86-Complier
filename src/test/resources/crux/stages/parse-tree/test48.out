program
  declarationList
    declaration
      functionDefinition
        parameterList
        type
        statementBlock
          statementList
            statement
              variableDeclaration
                type
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
            statement
              variableDeclaration
                type
            statement
              variableDeclaration
                type
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
            statement
              whileStatement
                expression0
                  expression1
                    expression2
                      expression3
                        expression0
                          expression1
                            expression2
                              expression3
                                designator
                          op0
                          expression1
                            expression2
                              expression3
                                literal
                statementBlock
                  statementList
                    statement
                      callStatement
                        callExpression
                          expressionList
                            expression0
                              expression1
                                expression2
                                  expression3
                                    designator
                    statement
                      callStatement
                        callExpression
                          expressionList
                    statement
                      variableDeclaration
                        type
                    statement
                      assignmentStatement
                        designator
                        expression0
                          expression1
                            expression2
                              expression3
                                designator
                            op1
                            expression2
                              expression3
                                designator
                    statement
                      assignmentStatement
                        designator
                        expression0
                          expression1
                            expression2
                              expression3
                                designator
                    statement
                      assignmentStatement
                        designator
                        expression0
                          expression1
                            expression2
                              expression3
                                designator
                    statement
                      assignmentStatement
                        designator
                        expression0
                          expression1
                            expression2
                              expression3
                                designator
                            op1
                            expression2
                              expression3
                                literal
